src/dash-table/DataTable.js

```diff
@@ -178,6 +178,31 @@ export const propTypes = {
             PropTypes.arrayOf(PropTypes.string)
         ]).isRequired,
 
+        number: PropTypes.shape({
+            presentation: PropTypes.oneOf(['input', 'dropdown']),
+            validation: PropTypes.shape({
+                allow_nan: PropTypes.bool,
+                on_change: PropTypes.oneOf([
+                    'nan',
+                    'passthrough',
+                    'prevent',
+                    'skip'
+                ])
+            })
```

Comment:
coercion / validation / reconciliation are affected by the validation settings

on_change defines how the table will behave if the value cannot be coerced into the destination type (number)
- nan will set the value NaN
- passthrough will set the value as-is without coercion (this provides a default behavior consistent with the current table behavior)
- skip will not set the value but will not prevent larger scope modifications (e.g. during a copy/paste)
- prevent will not set the value and will prevent larger scope modifications (the entire copy/paste operation will be cancelled if one cell's data is incompatible)